### ISSUE TITLE
Add Nginx Pagespeed support / install option

### ISSUE THREAD

**Issue by bery** [omega8cc/boa#882]
Hi,

I know that there has been some discussion in the past but I feel that having at least a discussion might be worth it:
- https://www.drupal.org/node/1256680#comment-7485550

In this post you are referring to multiple modules that can do similar things as Pagespeed module but:
- By design BOA should be used to simplify hosting and reduce maintenance times & costs
- Providing performance boost by default (same as letsencrypt integration for SSL) might be an excellent option
- Having at least the option to install pagespeed by default would be nice (I did not find a simple way how to do that without breaking BOA features)

The idea:
- Optionally compile Nginx with Pagespeed
- Update Pagespeed with Boa upgrades
- Nginx configs can be added manually on a per domain basis (similar to SSL proxy or CDN module)

Pros:
- optimized HTML and Assets (images, styles) by server configuration
- basically all [nginx pagespeed filters there are](https://developers.google.com/speed/pagespeed/module/filters)

Cons:
- resource consumption
- Some out-of-the box features can be broken (say by JS defer etc.)

If you do not turn pagespeed On, by default nothing will break - at least it will be easily installable and can make BOA even more awesome :)

**Comment by omega8cc** [comment]
I'm still not convinced it is worth the effort, especially in the BOA / Aegir context, where everything is designed from the Drupal multisite perspective and not single site with all optimizations moved to the Nginx level.

Again, this is how BOA with AdvAgg optimizes Drupal by default, so pagespeed test result is 92/100:

![92/100](http://s18.postimg.org/du9m34tbt/Page_Speed_Insights.jpg)

**Comment by omega8cc** [comment]
But if implementing this as an option would be simple, then why not?

**Comment by bery** [comment]
Awesome!

I am not saying that it would do everything for you, but for example from your screenshot - with one directive in pagespeed, you could optimize your images and get more, not speaking about other mobile filters and loads of others...

The [list](https://developers.google.com/speed/pagespeed/module/filters#categories-of-filters) is quite extensive and Google keeps adding more with every release.

**Comment by macmladen** [comment]
I think I can agree with original poster and maybe add that **pagespeed** is doing that on a lower level than Drupal modules can, that is server level, C written optimized versus complex PHP modules choking on Drupal.

It should be checked what is the efficiency of using multiple Drupal modules and optimization techniques versus **pagespeed** taking browser caching efficiency into account as well.

Additional **Pro** lies in originator, Google so **pagespeed** aligns you best with what search giant is expecting from you.

If I understood well, it may help even with non optimized Drupal site, that way we may help even those not too experienced to still build Drupal sites that perform well.

**Comment by flabel** [comment]
Last post was almost two years ago. Is this request still active and if so what is status?

**Comment by omega8cc** [comment]
@flabel It's scheduled to be included in the next major version 3.3.0 -- as tagged via milestone on the right here.

**Comment by bery** [comment]
That's great news! I'm looking forward to testing it out :)

**Comment by flabel** [comment]
Superb! Thank you for the quick reply.
